backend/src/main/scala/cromwell/backend/wdl/WriteFunctions.scala

```diff
@@ -30,19 +31,26 @@ trait WriteFunctions { this: WdlStandardLibraryFunctions =>
     }
   }
 
-  private def writeToTsv(functionName: String, params: Seq[Try[WdlValue]], wdlClass: Class[_ <: WdlValue with TsvSerializable]) = {
+  private def writeToTsv[A <: WdlValue with TsvSerializable](functionName: String, params: Seq[Try[WdlValue]], defaultIfOptionalEmpty: A): Try[WdlFile] = {
+    val wdlClass = defaultIfOptionalEmpty.getClass
+    def castOrDefault(wdlValue: WdlValue): A = wdlValue match {
+      case WdlOptionalValue(_, None) => defaultIfOptionalEmpty
+      case WdlOptionalValue(_, Some(v)) => wdlClass.cast(v)
+      case _ => wdlClass.cast(wdlValue)
+    }
+
     for {
       singleArgument <- extractSingleArgument(functionName, params)
-      downcast <- Try(wdlClass.cast(singleArgument))
+      downcast <- Try(castOrDefault(singleArgument))
       tsvSerialized <- downcast.tsvSerialize
       file <- writeContent(wdlClass.getSimpleName.toLowerCase, tsvSerialized)
     } yield file
   }
 
-  override def write_lines(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv("write_lines", params, classOf[WdlArray])
-  override def write_map(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv("write_map", params, classOf[WdlMap])
-  override def write_object(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv("write_object", params, classOf[WdlObject])
-  override def write_objects(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv("write_objects", params, classOf[WdlArray])
-  override def write_tsv(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv("write_tsv", params, classOf[WdlArray])
+  override def write_lines(params: Seq[Try[WdlValue]]): Try[WdlFile] = writeToTsv("write_lines", params, WdlArray(WdlArrayType(WdlStringType), List.empty[WdlValue]))
```

Comment:
Not for here since it would be in wdl4s, but it looks like some `X.empty` methods for `WdlArray`, `WdlMap` etc... would be convenient